### PR TITLE
[Security Hardened Shoot Cluster] Rule 2000 Implementation

### PR DESCRIPTION
**Pull request by georgibaltiev**
**What this PR does / why we need it**:
This PR is an implementation of the Rule 2000 of the Security Hardened Shoot Cluster Ruleset. It retrieves metadata about the shoot cluster and evaluates the EnableAnonymousAuthentication flag (if present) of the Kube API Server component.

**Which issue(s) this PR fixes**:
Part of #304 

**Special notes for your reviewer**:

**Release note**:

```feature user
Implementation for rule `2000` from the `security-hardened-shoot-cluster` ruleset for provider `garden`.
```

### CODE REVIEW

File: pkg/provider/garden/ruleset/securityhardenedshoot/rules/2000_test.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+		Expect(fakeClient.Create(ctx, shoot)).To(Succeed())
+		res, err := r.Run(ctx)
+		Expect(err).To(BeNil())
+		Expect(res).To(Equal(rule.RuleResult{RuleID: ruleID, RuleName: ruleName, Severity: severity, CheckResults: expectedCheckResults}))
+	},
+		Entry("should error when the shoot is not found",
+			func() { shoot.Name = "notFoo" },
+			[]rule.CheckResult{{Status: rule.Errored, Message: "shoots.core.gardener.cloud \"foo\" not found", Target: rule.NewTarget("kind", "Shoot", "name", "foo", "namespace", "bar")}},
+		),
+		Entry("should warn when kube-apiserver configuration is not set",
```
Comment (AleksandarSavchev) [inline pkg/provider/garden/ruleset/securityhardenedshoot/rules/2000_test.go:65]:
```suggestion
		Entry("should pass when kube-apiserver configuration is not set",
```

File: pkg/provider/garden/ruleset/securityhardenedshoot/rules/2000_test.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+			func() {},
+			[]rule.CheckResult{{Status: rule.Passed, Message: "Anonymous authentication is not enabled.", Target: rule.NewTarget()}},
+		),
+		Entry("should warn when anonymous authentication is not set",
```
Comment (AleksandarSavchev) [inline pkg/provider/garden/ruleset/securityhardenedshoot/rules/2000_test.go:69]:
```suggestion
		Entry("should pass when anonymous authentication is not set",
```

**Review by AleksandarSavchev — CHANGES_REQUESTED** [review]
Looks good, just 2 changes to test names.

File: pkg/provider/garden/ruleset/securityhardenedshoot/rules/2000_test.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+		}
+
+		r = &rules.Rule2000{
+			Client:         fakeClient,
+			ShootName:      shootName,
+			ShootNamespace: shootNamespace,
+		}
+	})
+
+	DescribeTable("Run cases", func(updateFn func(), expectedCheckResults []rule.CheckResult) {
```
Comment (AleksandarSavchev) [inline pkg/provider/garden/ruleset/securityhardenedshoot/rules/2000_test.go:54]:
```suggestion
	DescribeTable("Run cases", func(updateFn func(), expectedCheckResult rule.CheckResult) {
```
Last comment, no need for `expectedCheckResults` to be a slice when the function can only return a single `rule.CheckResult`

**Review by AleksandarSavchev — APPROVED** [review]
/lgtm

File: pkg/provider/garden/ruleset/securityhardenedshoot/rules/2000.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+		return rule.Result(r, rule.ErroredCheckResult(err.Error(), rule.NewTarget("name", r.ShootName, "namespace", r.ShootNamespace, "kind", "Shoot"))), nil
+	}
+
+	switch {
+	case shoot.Spec.Kubernetes.KubeAPIServer == nil || shoot.Spec.Kubernetes.KubeAPIServer.EnableAnonymousAuthentication == nil:
+		return rule.Result(r, rule.PassedCheckResult("Anonymous authentication is not enabled.", rule.NewTarget())), nil
+	case *shoot.Spec.Kubernetes.KubeAPIServer.EnableAnonymousAuthentication:
+		return rule.Result(r, rule.FailedCheckResult("Anonymous authentication is enabled on the kube-apiserver.", rule.NewTarget())), nil
+	default:
+		return rule.Result(r, rule.PassedCheckResult("Anonymous authentication is disabled on the kube-apiserver.", rule.NewTarget())), nil
```
Comment (dimityrmirchev) [inline pkg/provider/garden/ruleset/securityhardenedshoot/rules/2000.go:52]:
```suggestion
	case *shoot.Spec.Kubernetes.KubeAPIServer.EnableAnonymousAuthentication:
		return rule.Result(r, rule.FailedCheckResult("Anonymous authentication is enabled for the kube-apiserver.", rule.NewTarget())), nil
	default:
		return rule.Result(r, rule.PassedCheckResult("Anonymous authentication is disabled for the kube-apiserver.", rule.NewTarget())), nil
```

**Review by dimityrmirchev — CHANGES_REQUESTED** [review]
Just a single wording change request.

**Review by dimityrmirchev — APPROVED** [review]
/lgtm

**Pull request merged**